### PR TITLE
Replace blob chain info query usage with blob download

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Now that we can download blobs, using the chain info query isn't needed anymore

## Proposal

Use the download blob entrypoint instead

## Test Plan

CI

### CODE REVIEW

**Comment by andresilva91** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2079?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2079** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2079?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2074** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2074?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: linera-core/src/client.rs [hunk 1]
```diff
@@ -748,15 +748,17 @@ where
...
-        }))
+        future::join_all(
+            blob_ids
+                .iter()
+                .map(|blob_id| LocalNodeClient::<S>::try_download_blob(nodes.to_owned(), *blob_id)),
+        )
         .await
         .into_iter()
         .flatten()
+        .flatten()
```
Comment (afck) [inline linera-core/src/client.rs:761]: Shouldn't we fail if one of the blobs can't be found?

File: linera-core/src/local_node.rs [hunk 2]
```diff
@@ -71,6 +72,9 @@ pub enum LocalNodeError {
 
     #[error("The chain info response received from the local node is invalid")]
     InvalidChainInfoResponse,
+
+    #[error("Got different blob id from downloaded blob when trying to download {blob_id:?}")]
```
Comment (afck) [inline linera-core/src/local_node.rs:76]:
```suggestion
    #[error("Got different blob ID from downloaded blob when trying to download {blob_id:?}")]
```

File: linera-core/src/local_node.rs [hunk 3]
```diff
@@ -662,42 +655,39 @@ where
...
     {
         // Sequentially try each validator in random order.
         validators.shuffle(&mut rand::thread_rng());
-        for (name, mut node) in validators {
-            if let Some(blob) =
-                Self::try_download_blob_from(name, &mut node, chain_id, blob_id).await
-            {
-                return Some(blob);
+        for mut node in validators {
+            if let Some(blob) = Self::try_download_blob_from(&mut node, blob_id).await? {
```
Comment (afck) [inline linera-core/src/local_node.rs:668]: We shouldn't fail if one faulty/malicious validator sends us the wrong blob. We should continue trying to download it from someone else.

**Comment by afck** [comment]
Closing in favor of #2098.

**Pull request closed without merging**